src/e2e/java/teammates/e2e/pageobjects/PageWithNotificationBanner.java

```diff
@@ -0,0 +1,85 @@
+package teammates.e2e.pageobjects;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+
+/**
+ * Page Object Model for page that can display notification banners.
+ */
+public class PageWithNotificationBanner extends AppPage {
+    @FindBy(tagName = "tm-notification-banner")
+    private WebElement bannerContainer;
+
+    @FindBy(id = "btn-close-notif")
+    private WebElement closeNotifButton;
+
+    @FindBy(id = "btn-mark-as-read")
+    private WebElement markNotifAsReadButton;
+
+    public PageWithNotificationBanner(Browser browser) {
+        super(browser);
+    }
+
+    @Override
+    protected boolean containsExpectedPageContents() {
+        // Notification Banner can appear on any page except for the past notification page
+        String pageTitle = getPageTitle();
+        return !"Instructor Notifications".equals(pageTitle) || !"Student Notifications".equals(pageTitle);
+    }
+
+    public void verifyNotificationBannerIsNotVisible() {
+        List<WebElement> bannerContent = bannerContainer.findElements(By.id("banner-contents"));
+        assertEquals(0, bannerContent.size());
+    }
+
+    public void verifyNotificationBannerIsVisible(NotificationAttributes notification) {
+        List<WebElement> bannerContent = bannerContainer.findElements(By.id("banner-contents"));
+        String title = bannerContainer.findElement(By.tagName("h5")).getText();
+        String message = bannerContainer.findElement(By.className("banner-text")).getText();
+
+        assertEquals(1, bannerContent.size());
+        assertEquals(notification.getTitle(), title);
+        assertEquals(removeParagraphTag(notification.getMessage()), message);
+    }
+
+    public String removeParagraphTag(String message) {
+        return message.replace("<p>", "").replace("</p>", "");
+    }
+
+    public void clickCloseNotificationBannerButton() {
+        waitForElementToBeClickable(closeNotifButton);
+        click(closeNotifButton);
+        waitUntilAnimationFinish();
+    }
+
+    public void clickMarkAsReadButton() {
```

Comment:
Looking at the fact that the notification banner spans across almost all pages except student/instructor notifications page and admin pages, I have a feeling we might need to make `verifyNotificationBannerIsNotVisible()`, `clickCloseNotificationBannerButton()` and `clickMarkAsReadButton()` accessible to all pages i.e. make them part of `AppPage`. However I'm not entirely sure whether this is a good idea. If it is done this way, then
1. Tests for the behaviour of the reusable notification banner (verify correct notification details, close notification, mark as read) will be done on an arbitrary page, like student home page used in this case, where the notification banner can appear on.
2. Tests for the absence of notification banner on user notifications page can be done quite easily in `StudentNotificationsPageE2ETest` and `InstructorNotificationsPageE2ETest` respectively.
3. Tests for presence of notification banner can be done in important pages where we expect users to first see the notifications, like instructor and student home page i.e. in `StudentHomePageE2ETest` and `InstructorHomePageE2ETest` respectively.

Reply:
@wkurniawan07 what do you think? The problem we are facing is that the notification banner does not belong solely to any page, and testing it's behaviour could involve multiple pages like what @NicolasCwy was trying to do by switching between tabs in the navbar to check for presence/absence of the notification banner.